### ISSUE TITLE
高次多項式の圧力勾配誤差問題

### ISSUE THREAD

**Issue by maru10t** [openmps/openmps#49]
#48 で 圧力分布が p(x) = a x^3 のような三次多項式(四次でも確認)のとき、
圧力勾配 の 解析解からの相対誤差が、解像度を上げても収束しない問題 (~1%の有限値となる)

**Comment by maru10t** [comment]
#48 では、テストでの許容誤差上限を5%に引き上げて一時的に対処した

**Comment by maru10t** [comment]
めも
* 現在は等間隔の正方格子に粒子を配置している。粒子配置に依存する問題である気がする。

**Comment by LWisteria** [comment]
現在の圧力勾配モデルが(pj-pi)/rであることを考慮すると、単純な1次式なのでどうやっても高次元は再現できないのでは・・・という気がしました。

**Comment by maru10t** [comment]
単調収束性(?)のような適切な離散化であれば、1次式であっても高次多項式の微分は再現できると思いました。
例: 格子法で差分を(p_{i+1}-p_{i})/⊿x という1次近似で格子数を増やして⊿x->0とすれば真の微分値に近づく

**Comment by LWisteria** [comment]
そうですね。例えばr_e=2l_0にすると、1次元方向で考えれば完全に差分法と一致するはずなので、適合性はあって収束しそうな気はしますが・・・。
